apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -503,12 +503,12 @@ func (api *ProvisionerAPI) AvailabilityZone(ctx context.Context, args params.Ent
 		}
 		machineUUID, err := api.machineService.GetMachineUUID(ctx, coremachine.Name(tag.Id()))
 		if err != nil {
-			result.Results[i].Error = apiservererrors.ServerError(machineerrors.MachineNotFound)
```

Comment:
Fixing the returned `machineerror` over the wire.

Reply:
Can we do this instead:

```sh
result.Results[i].Error = apiservererrors.ServerError(errors.Errorf("%w: %w", err, NotFound))
```

Reply:
Doesn't `ServerError` just get the cause and hide the error in a code? I guess there's no harm in doing this, but I don't think anyone would see that error.